plugins/modules/foreman_host_power.py

```diff
@@ -87,41 +92,34 @@
     sample: "off"
  '''
 
-try:
-    from nailgun.entities import (
-        Host,
-    )
-
-    from ansible.module_utils.ansible_nailgun_cement import (
-        find_host,
-        ForemanEntityAnsibleModule,
-        naildown_power_state,
-        query_power_state,
-    )
-except ImportError:
-    pass
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule, parameter_entity_spec
 
 
 def main():
     module = ForemanEntityAnsibleModule(
-        argument_spec=dict(
-            hostname=dict(required=True),
-            state=dict(default='present', choices=['on', 'off', 'state']),
+        entity_spec=dict(
+            name=dict(aliases=['hostname'], required=True),
+            state=dict(default='state', choices=['on', 'start', 'off', 'stop', 'soft', 'reboot', 'cycle', 'reset', 'state', 'status']),
         ),
     )
 
-    (host_dict, state) = module.parse_params()
+    entity_dict = module.clean_params()
 
     module.connect()
 
-    entity = find_host(module, host_dict['hostname'], failsafe=True)
-
-    if state == 'state':
-        power_state = query_power_state(module, entity)
-        module.exit_json(changed=False, power_state=power_state)
+    entity = module.find_resource_by_name('hosts', name=entity_dict['name'], failsafe=False, thin=True)
 
+    params = {'id': entity_dict['name']}
+    _, power_state = module.resource_action('hosts', 'power_status', params=params)
```

Comment:
I guess not...

Reply:
hmm, this mobile client dosn't seem to comment on quite the right line. Glad you understood me anyway.